src/interfaces/protocols/types.ts

```diff
@@ -11,7 +11,10 @@ export type ProtocolsConfigureDescriptor = {
 
 export type ProtocolDefinition = {
   labels: {
```

Comment:
This is non-blocking, just discussion: calling this `labels` now feels like an over-simplification, maybe `constraints` is a better name.

Reply:
Agreed, especially as we add more to `labels`. I'm open to changing this field (though not in the PR preferably).

`constraints` doesn't feel right either. `constraints` implies to me that we only need to list a label/type/name/whatever here if we have specific restrictions to put on it like `dataFormat`. But in fact we need to include all labels/types/names/whatevers in this property if we want them to be valid in the `record` section.

Spitballing some more ideas:
* `recordTypes`
* `types`
* `models`
* `tables`